packages/selenium-ide/src/neo/IO/filesystem.js

```diff
@@ -72,7 +72,18 @@ export function loadAsText(blob) {
 
 export function saveProject(_project) {
   const project = _project.toJS()
-  downloadProject(project)
+  if (UiState.isControlled) {
+    const saveMessage = {
```

Comment:
That is not what I mean, can we move this to original place in 
 ? Promise.resolve()
And please make this send event a separated method.

Reply:
If we move to original place, then we get 2 requests on save: one from plugin emit and one from this message. Tomer said plugin emit should not happen in control mode